test/unit/SingleSignatureDirtyHack.t.sol

```diff
@@ -0,0 +1,386 @@
+        uint256 gasLimit,
+        address to,
+        uint256 amount,
+        uint256 deadline,
+        uint256 nonce,
+        uint256 requestedAmount,
+        bytes32 salt,
+        address operator,
+        address accessedAcount,
+        bytes32[3] memory slots
+    ) public {
+        bytes memory extraData;
+        {
+            AccessListElem[] memory accessList = new AccessListElem[](1);
+            accessList[0] = AccessListElem({account: accessedAcount, slots: new bytes32[](3)});
+            accessList[0].slots[0] = slots[0];
+            accessList[0].slots[1] = slots[1];
+            accessList[0].slots[2] = slots[2];
+            extraData = abi.encode(gasPriorityPrice, accessList);
+        }
+
+        _commonTest(
+            signerPk,
+            gasPrice,
+            gasLimit,
+            to,
+            amount,
+            deadline,
+            nonce,
+            requestedAmount,
+            salt,
+            operator,
+            extraData,
+            _eip1559Payload,
+            _transferEIP1559
+        );
+    }
+
+    function _commonTest(
+        uint256 signerPk,
+        uint256 gasPrice,
+        uint256 gasLimit,
+        address to,
+        uint256 amount,
+        uint256 deadline,
+        uint256 nonce,
+        uint256 requestedAmount,
+        bytes32 salt,
+        address operator,
+        bytes memory extraData,
+        function(uint256, uint256, uint256, address, bytes memory, bytes memory) view returns (bytes32) getPayload,
+        function(address, SingleSignatureDirtyHack.TransferParams memory, uint256, uint256, PackedSignature memory, bytes memory) internal
+            transfer
+    ) internal {
+        signerPk = bound(signerPk, 2, 115792089237316195423570985008687907852837564279074904382605163141518161494336);
+        gasLimit = bound(gasLimit, 0, 30_000_000);
+        deadline = bound(deadline, block.timestamp, type(uint256).max - 1);
+        nonce = bound(nonce, 1, type(uint64).max - 3);
+        amount = bound(amount, 2, type(uint256).max);
+        requestedAmount = bound(requestedAmount, 1, amount);
+
+        (bytes32 structHash, PackedSignature memory sig) =
+            _signPayload(salt, operator, deadline, nonce, gasPrice, gasLimit, amount, signerPk, extraData, getPayload);
+
+        SingleSignatureDirtyHack.TransferParams memory transferParams = SingleSignatureDirtyHack.TransferParams({
+            structHash: structHash,
+            token: token,
+            from: vm.addr(signerPk),
+            to: to,
+            amount: amount,
+            nonce: nonce,
+            deadline: deadline,
+            requestedAmount: requestedAmount
+        });
+
+        vm.prank(transferParams.from);
+        token.approve(address(harness), amount);
+        deal(address(token), address(transferParams.from), amount);
+
+        transfer(operator, transferParams, gasPrice, gasLimit, sig, extraData);
+
+        // can use incremental nonce
+        nonce++;
```

Comment:
why is `nonce == 1` outside the scope of the test?

Reply:
this test is for proving that after a valid txn doing another with a greater nonce works. Notice that before this line a transfer was already accepted so only nonces from `nonce + 1` are valid